femwell/maxwell/waveguide.py

```diff
@@ -148,6 +150,28 @@ def calculate_pertubated_neff(self, delta_epsilon):
             * 0.5
         )
 
+    def calculate_intensity(self) -> Tuple[NDArray, Basis]:
+        """Calculates the intensity of a mode.
+
+        The intensity is calculated from the cross-product between the electric and magnetic field, as 
+        described in https://doi.org/10.1364/OE.16.016659.
+
+        The calculation is performed as follows:
+        1) The electric and magnetic fields are interpolated on the quadrature points with the simulation basis;
+        2) The intensity is calculated directly on the quadrature points;
+        3) The intensity is projected on a new discontinuous, piecewise linear triangular basis.
+
+        Returns:
+            NDArray, Basis: Intensity and plot-ready basis
+        """
+        (Ex, Ey), _ = self.basis.interpolate(self.E)
+        (Hx, Hy), _ = self.basis.interpolate(np.conj(self.H))
+        intensity = 0.5 * np.real(Ex * Hy - Ey * Hx)
+        basis2 = self.basis.with_element(ElementDG(ElementTriP1()))
+        intensity2 = basis2.project(intensity)
+
+        return intensity2, basis2
```

Comment:
Up to now, we always return first the basis (see for example https://github.com/HelgeGehring/femwell/blob/main/femwell/thermal.py#L91)
Would be great to stay consistent here!

Reply:
Sure!